10up/classifai
# Add plugin debug information to the Site Health screen

This resolves #83, which describes a need to display plugin debug information on the new Site Health screen introduced to core in 5.2.

### Description of the Change

It adds a new accordion drawer to the Site Health info screen that looks like this: 

![classifai-site-health](https://user-images.githubusercontent.com/9020968/61959886-fbb78700-af89-11e9-94fe-55ded5adc7da.png)

We do this by hooking into the new `debug_information` filter provided by core, adding a key for `classifai`, and providing a set of plugin details that would be useful for debugging purposes. This is done via a `classifai_debug_information` filter that can be hooked into throughout the plugin.

### Alternate Designs



- The `DebugInfo` class provides the `classifai_debug_information` filter and delegates the adding of debug information to the individual features in the plugin via the filter. Alternatively, we could do all the work needed to retrieve debug information in the `DebugInfo` class. This might be simpler, but I liked the extensibility the filter provides. 
- Currently the IBM Watson features display as a JSON encoded string. These could be broken out into their own lines. I just didn't see a clean way to do it within the limited structure of the Site Health drawer.

### Benefits



Plugin users have a concrete, nontechnical step to share information needed for debugging.

### Possible Drawbacks



Expecting debug information to be provided with all plugin features adds a small amount of overhead.

### Verification Process

Using an up-to-date Version of WP: 

1. In the admin, navigate to Tools/Site Health and switch to the "Info" tab.
2. Find and open the "ClassifAI" accordion drawer.
3. Verify that the information shown matches your environment.
4. Verify the same in a different environment with different settings, or modify your settings and re-check.

### Checklist:






### Applicable Issues

#83
